components/sup/src/sys/unix/abilities.rs

```diff
@@ -16,20 +16,32 @@
 //! abilities based on querying the available Linux capabilities; the
 //! functioning of the Supervisor may change depending on the answer.
 
-use caps::{self, Capability, CapSet};
+pub use self::imp::*;
 
-/// This is currently the "master check" for whether the Supervisor
-/// can behave "as root".
-///
-/// All capabilities must be present. If we can run processes as other
-/// users, but can't change ownership, then the processes won't be
-/// able to access their files. Similar logic holds for the reverse.
-pub fn can_run_services_as_svc_user() -> bool {
-    has(Capability::CAP_SETUID) && has(Capability::CAP_SETGID) && has(Capability::CAP_CHOWN)
+#[cfg(target_os = "linux")]
+mod imp {
+    use caps::{self, Capability, CapSet};
+
+    /// This is currently the "master check" for whether the Supervisor
+    /// can behave "as root".
+    ///
+    /// All capabilities must be present. If we can run processes as other
+    /// users, but can't change ownership, then the processes won't be
+    /// able to access their files. Similar logic holds for the reverse.
+    pub fn can_run_services_as_svc_user() -> bool {
+        has(Capability::CAP_SETUID) && has(Capability::CAP_SETGID) && has(Capability::CAP_CHOWN)
+    }
+
+    /// Helper function; does the current thread have `cap` in its
+    /// effective capability set?
+    fn has(cap: Capability) -> bool {
+        caps::has_cap(None, CapSet::Effective, cap).unwrap_or(false)
+    }
 }
 
-/// Helper function; does the current thread have `cap` in its
-/// effective capability set?
-fn has(cap: Capability) -> bool {
-    caps::has_cap(None, CapSet::Effective, cap).unwrap_or(false)
+#[cfg(target_os = "macos")]
```

Comment:
How does this come into play, given that we're not running (or even building) the supervisor on MacOS?

Reply:
I may have been the last one holding out, but up until a few days ago I was still using `cargo check` from MacOS because I never replaced my docker container with a Linux VM. This is specifically there as a shortest route to allowing cargo check to run.

Reply:
I think you were the last holdout; I don't know that anyone's doing dev directly on MacOS anymore.

Having extra "macos" config flags for that seems like it'd get confusing very quickly.

Reply:
Of course I can remove these, but just by doing a bit of work that needs cleaning up. It's been our vision for Habitat to run on any OS. I don't think we should shy away from incremental changes to get us there.

Reply:
I agree about making it run everywhere... as long as we've got an issue filed for taking a pass through the OS-specific code and tightening it up, I'm cool with incremental tweaks. 

Maybe a comment on the new MacOS-specific code that seems crazy (since we don't build the Supervisor there yet, for instance), with a link to that issue would be a good intermediate step, just so anyone looking through the code will have a pointer to the long-term plans?